lib/Analysis/Allocation.cpp

```diff
@@ -131,9 +140,8 @@ class AllocationAnalysis {
       auto dstTy = cvtLayout.result().getType().cast<RankedTensorType>();
       auto srcEncoding = srcTy.getEncoding();
       auto dstEncoding = dstTy.getEncoding();
-      if (srcEncoding.isa<SharedEncodingAttr>() ||
-          dstEncoding.isa<SharedEncodingAttr>()) {
-        // Only blocked -> blocked conversion requires for scratch allocation
+      if (srcEncoding.isa<SharedEncodingAttr>()) {
+        // only block->block and block->shared is supported now
```

Comment:
We shouldn't revise here in order to support block->shared.
blocked->shared doesn't requires for "scratch" memory, and it should has already been taken care of.